### DISCUSSION TITLE
Windows installing Jekyll error - please help

### DISCUSSION THREAD

**Discussion by XebbyLIS** [CollectionBuilder/collectionbuilder.github.io#109] [Q&A]
Hi, I'm new to all of this. I have installed everything based on default settings and locations except when specified otherwise in the instructions for software. 

I'm using Windows 11 and having issues with installing Jekyll. I had to uninstall and reinstall ruby twice for it to work, but at least when I run this command in Git Bash I get an answer:
_$ruby -v
ruby 3.2.2 (2023-03-30 revision e51014f9c0) [x64-mingw-ucrt]_

Specifically, when i run this command in Git Bash, i get this error. 
_$ jekyll -v
/c/Users/hella/AppData/Local/Microsoft/WindowsApps/jekyll: line 6: /c/Users/hella/AppData/Local/Microsoft/WindowsApps/ruby: No such file or directory_

I've tried uninstalling and reinstalling ruby, entering the commands on the windows command prompt, and searching but I have no idea where to continue troubleshooting. First image is what shows up when i open the terminal in VS Code 
![Screenshot 2023-12-18 213223](https://github.com/CollectionBuilder/collectionbuilder.github.io/assets/154286166/00a2aee0-6337-4753-b5a6-17ae8fd0ffe1)

![Screenshot 2023-12-18 211440](https://github.com/CollectionBuilder/collectionbuilder.github.io/assets/154286166/e3cd5a5c-1d65-413b-8b0a-8b4d9e784817)
![Screenshot 2023-12-18 211413](https://github.com/CollectionBuilder/collectionbuilder.github.io/assets/154286166/ea417adf-77c4-40e3-a5dc-e32f3f874f42)
![Screenshot 2023-12-18 174301](https://github.com/CollectionBuilder/collectionbuilder.github.io/assets/154286166/2cbd5774-aa9b-4aed-94f5-23431556a10e)
![Screenshot 2023-12-18 212923](https://github.com/CollectionBuilder/collectionbuilder.github.io/assets/154286166/8a7406dd-7491-45fa-8fd9-4a9f88900e7f)

**Comment by evanwill** [comment]
@XebbyLIS from your screen shots, it *looks like* you got git, ruby, bundler, and jekyll installed. 
One issue to check first, did you fully close out your VS Code and Git Bash terminal windows? Sometimes your "path" needs to be refreshed by a reload of the shell. VS Code has been making that more difficult by using “persistent sessions” feature which can cause a lot of mysterious issues. 

So first I would go into VS Code > Settings, search the settings for "persistent" then uncheck the option “Terminal > Integrated: Enable Persistent Sessions” and also choose "never" in the option "Terminal > Persistent Session Revive Process".
Then close the VS Code terminal with the "trash" icon. 
Restart your computer for good measure. 
And try again? 

One other check that might help debug, is to open a "Git Bash" terminal directly (i.e. not the one integrated in VS Code, just search "Git Bash" in your start menu). Check `jekyll -v` there to see if it is installed. If jekyll is working there, it would help determine there is something weird with your VS Code. 

The other thing I noticed in your Terminal screenshots, it looks like your GitHub folder is in your Onedrive folder. That can also cause strange issues, as Onedrive tries to sync all the files that jekyll creates. It would be better to try to move your repositories (i.e. the folders in the "GitHub" folder directly on to your C drive rather than in Onedrive. 

Let us know how it goes, and maybe someone else with Windows 11 will chime in!

**Reply by XebbyLIS** [reply]
things I had already tried:
-restarting computer, VS Code, and Git Bash
-turned off persistent session
-checked Jekyll -v in Git Bash

I had hoped it wouldn't be because of one drive, because I want to make sure I move all necessary files to the new location. OneDrive is so sneaky, I didn't realize all my documents were actually stored in OneDrive because the file path is obscured by how windows organizes things. 

-manually moving the relevant folders out of OneDrive didn't help, time to reinstall everything from scratch. I think i'd be helpful to update the install instructions to state that OneDrive will mess it all up.

**Reply by evanwill** [reply]
Let us know if moving out of Onedrive ends up helping... There is an alert on the [clone repository](https://collectionbuilder.github.io/cb-docs/docs/repository/clone/#clone-your-new-repository) section about it, but I guess we could make that stronger if this ends up being the solution. 

In Git Bash what does `jekyll -v` give you? The same error? 

I don't have Windows 11 computer to test on, but we had a bunch of fresh installs during a workshop a few weeks ago and didn't run into any issues--so not sure... ruby install is the worst part.

**Reply by XebbyLIS** [reply]
jekyll -v in both VS Studio code and Git Bash lead to the same error message. (the first image with the "line 6:" and "No such file or directory" 
Moving out of Onedrive just by clicking and dragging did not work (I have never installed something like Jekyll so perhaps there was additional items to move) so I ended up uninstalling and reinstalling it all. 

Even after uninstalling everything, I got a similar error- meaning the way the default installation for either Jekyll or Ruby still follows the Microsoft path- it isn't just One Drive. 

I'm guessing, however I believe the issue is related to windows "C:\ProgramData\Microsoft" as a storage location that is considered "default"

I ended up just uninstalling One Drive from my PC entirely, since i do regular manual backups anyhow. I didn't know how much Windows 11 was obfuscating when you are saving to One Drive. They display shorted file locations that conveniently leave out the /One Drive/ unless you examine the path manually. They also use the same icons for documents, desktop, ect., from prior versions of windows. 
When I deleted One Drive (after dragging existing files onto my C: drive) I effectively deleted my entire desktop icon arrangement.  I'm not a tech expert but I did work in PC sales around 5 years ago and learned a lot at that time. In other words, If I was having issues knowing that things were being saved to One Drive (as windows forces this save path as a default) I would assume others will have the same issue. 

For the prior installation I was having error messages with: 
I did notice that Git Bash installed properly [ C:\Program Files\Git ] without any connection to that microsoft folder.  Meanwhile Ruby installed onto the window's folder automatically. I probably only needed to reinstall Ruby and Jekyll but I chose to reinstall everything for ease of use.

**Comment by XebbyLIS** [comment]
After more troubleshooting (Git bash and VS studio code gave me the same error again) I uninstalled and reinstalled Visual Studio Code. I don't know enough about computers to know if this was necessary, but I wanted to download it to the same C:\Program Files\ location because that's where Git and Ruby are installed. This time installed the VSCodeSetup instead of the default VSCodeUserSetup installer (if you try to install the VSCodeUserSetup installer in this location of C:\Program Files)

This got more errors than last time. It only installed one gem. and when I check for the installation using  jekyll -v I get the same error as before. 
Same error happens in VS studio code [bash] as Git 
This is using the most recent ruby update 3.3.0

![Screenshot 2023-12-28 171306](https://github.com/CollectionBuilder/collectionbuilder.github.io/assets/154286166/5e30f868-06bd-458b-bd52-f161eadac64c)


I had to install the System Installer Version of VS Code. Even then, I get the same error- for whatever reason windows is searching for ruby and jekyll in C:\Users\hella\AppData\Local\Microsoft\WindowsApps\ruby and C:\Users\hella\AppData\Local\Microsoft\WindowsApps\jekyll 

the ruby folder does not exist in that location. Jekyll exists there but not as a file. 
when i loaded C:\Users\hella\AppData\Local\Microsoft\WindowsApps\jekyll I used firefox to open it which generated the following:

![image](https://github.com/CollectionBuilder/collectionbuilder.github.io/assets/154286166/fc7f3707-e6ae-4868-a27c-c1d1ce151c6e)

I have tried searching in my files and verified that these are not blank folders labeled with ruby or jekyll- they do not exist on my PC. 

At this point I'm assuming i either need to download things into particular file paths even if they are likely to cause issues OR change how Git or VS Studio Code or something else is searching for the relevant files. Because it does not have the correct file path. 

I also noticed that Github Desktop does not let me choose where to install it- it just installs. 

I am going to be teaching other grad students how to install all of this to practice using Collection Builder. 
Does anyone know what the best windows file paths are for installing all of the programs? 
All the instructions say to install using default file paths but that still creates this error (and windows default paths are related to One Drive). 

While i could install everything under my C:\Program Files\ area with specific file names for programs, but if a student does not have administrative access they won't be able to install the VSCode there, as they would be using the VSCodeUserSetup installer. Then again, I can't even get this to work, so I have no idea. 

I'm willing to try more troubleshooting, however I am out of the default options offered by the tutorials. 

I'm gonna try downloading ruby directly into the file location it keep referring to as well as building a folder labeled jekyll to see if that fixes these issues

**Comment by XebbyLIS** [comment]
I tried troubleshooting by creating and adding a ruby folder to C:\Users\hella\AppData\Local\Microsoft\WindowsApps  
so the ruby folder exists now. 

I tried adding a jekyll folder as well but it was never filled with anything so i deleted it. 

More installation happened however I now have a new error when i type jekyll -v in Git bash or in VSCode [Bash]
/c/Users/hella/AppData/Local/Microsoft/WindowsApps/jekyll: line 6: /c/Users/hella/AppData/Local/Microsoft/WindowsApps/ruby: Is a directory
/c/Users/hella/AppData/Local/Microsoft/WindowsApps/jekyll: line 6: exec: /c/Users/hella/AppData/Local/Microsoft/WindowsApps/ruby: cannot execute: Is a directory

**Comment by XebbyLIS** [comment]
I have now spent 2 hours trying things via chat GPT- 
I get the same successful appearing installation message for gem install jekyll bundler 
and then the same error 
$ jekyll -v
/c/Users/hella/AppData/Local/Microsoft/WindowsApps/jekyll: line 6: /c/Users/hella/AppData/Local/Microsoft/WindowsApps/ruby: No such file or directory


I tried following chat GPT to see if I could change the path that my PC is choosing to search for jekyll. Trying the options Chat GPT offered did not work either. I even made a  "C:\devtools" folder to try to forcibly re-route it. It looks like it did install there! But whenever i type 
jekyll -v in Git bash it returns the same error as earlier. 
![image](https://github.com/CollectionBuilder/collectionbuilder.github.io/assets/154286166/59f48cf8-0731-48b7-8f91-1c0c3d78508e)


I am at a loss. I double-checked and cleared additional files containing jekyll or ruby. I've been restarting Git and restarting my computer for good measure. 

I don't know why I cannot seem to control where jekyll is being retrieved from or why it does not match with the place it is being installed. Chat GPT suggested I make a .bashrc file that contained the following lines of code
export GEM_HOME=/c/devtools
export GEM_PATH=/c/devtools

even with that saved, it does not work. I cannot get Git bash (both by itself and on VS Code) to show jekyll -v as installed with a version. 
ruby -v returns the proper response.

**Reply by dcnb** [reply]
Hey @XebbyLIS! Could we schedule a time for next week where we can zoom and see what's going on more? You can email us at collectionbuilder.team@gmail.com and we can set up a zoom. I'm sorry you're having so many difficulties!

**Reply by XebbyLIS** [reply]
hi @dcnb ! See below for details on how I finally got it to work!

I think the main issue was related to following suggested installation paths on a windows PC and my inexperience with this type of software. It seems like once a path was chosen, windows kept insisting on using this path.

Basically, Jekyll was getting installed this whole time, but it seems like it was not being retrieved from where i installed it. So even though I thought I uninstalled everything related to Jekyll and Ruby before reinstalling, despite adding new file locations for them to be saved and installed *to*, Git was trying to retrieve the files from where i had installed them initially (through windows apps which as a file location is not compatible at all.) 

Because i'm a TA for a digital libraries class this semester I followed the installation instructions on a Mac - troubleshooting was far easier. I got everything installed on my brothers macbook in under 45 minutes. I would love to truly understand what went awry with my installation (aside from an hour I probably lost due to chat GPT or myself misplacing a ~ when creating the file path for the .bashrc file) 

My best guess is that the problem was related to where something was initially installed: didn't matter if i re-installed it, Git Bash was still searching in that exact same area. The mere accident of installing it on OneDrive (because I wanted to follow default installation paths- I tend to take instructions very literally) made everything harder. This may have been exacerbated by the fact that I've been using VS Studio code for 8 months already. Because it is Microsoft Visual Studio Code, perhaps that is why the suggested installation location is on windows apps- it assumes compatibility. 

Granted, if you wanna look over how I fixed the issue and offer professional coding insight on what would have made this easier _after_ i'd already installed it on OneDrive, I would love to hear it! the response after this includes my chat GPT correspondence and my best guesses. 

Most importantly, I would appreciate knowing where a software expert would suggest to install Ruby as well as Jekyll. As the default location windows suggests causes errors, I know I will tell students to install NOT on One Drive or through anything connected to /microsoft/WindowsApps. That said, I don't have a confident alternative location to suggest to others. 

I am assuming that because I got jekyll to finally give a response in Git Bash with Jekyll -v, going forward with the rest of the CSV tutorial should work as intended. Fingers crossed.

**Comment by XebbyLIS** [comment]
I finally got it to work after several hours of troubleshooting with chat GTP. 
I hope this is just because I initially installed it following the "default" path - it seems like windows was determined to try to use that /c/Users/hella/AppData/Local/Microsoft/WindowsApps/jekyll location even after i had uninstalled all relevant programs. 

Some errors also happened because I had to manually open every installer and file in Admin Mode (which is not enabled by default).

One thing that exacerbated the issue was "a Ruby executable in a directory that's conflicting with your desired installation location." 
Even after I had searched for all mentions of ruby or jekyll on my C: drive, something had avoided detection until i checked environmental variables. 
Chat GPT instructions:
"Open the System Properties on your computer, go to the "Advanced" tab, and click on the "Environment Variables" button. Check both the "User variables" and "System variables" sections for any references to Ruby or unwanted directories."

Chat GPT Log: 
https://chat.openai.com/share/7e984e60-efe4-48f5-b355-c48d3879b170

**Comment by dcnb** [comment]
@XebbyLIS we're very glad you were able to figure this out! Sorry it was such a pain! I do think it had something to do with OneDrive and your getting started in the wrong directory. I don't know that I can add any more context or code-based advice. I'm going to talk to the rest of the team and have them look at this and then add something in the docs to help catch others who might be having issues. 

If you can highlight any sections in the docs that would have helped you, let us know. Otherwise, we'll put something in there and let you know what we did. Happy New Year!

**Comment by XebbyLIS** [comment]
Even doing all of this lead to eventual errors- it seems like git still searches for things in that one drive folder. 
I ran into more walls when i tried to do the bundle installer. I tried working with Chat GPT to find all the locations that ruby, git, VS studio code, and jekyll were installed on my pc. If anyone has resources about all the places that windows 11 installs parts of these, that would be helpful. It seems like the path things are installed STICKS even if everything is installed via the windows installer- in other words: it isnt a clean install

 The hardest part is that uninstalling and reinstalling everything doesn't change the fact that root file locations are set up somewhere else (i think that's the issue). Basically, if you unintentionally install things on One Drive, undoing the issue is hard. 

I'm going to attempt to reinstall everything once more by creating another "user" on my windows laptop. If that doesn't solve the problem i'm gonna reset my PC and then START by using the "uninstall windows bloatware" and "uninstall onedrive" options that exist on github. I think i understand why so many programmers have a linux system.

**Comment by XebbyLIS** [comment]
No luck, still having issues installing on a different username.
I expect my issue is related to the fact that I initially installed VS Studio Code a year ago and didn't realize the path included OneDrive. So i'm resetting my PC because this issue seems to be related to all the windows bloatware.

Resetting my entire laptop and ensuring OneDrive was no longer installed did not fix the issues. I have sent an email requesting help.

**Comment by ShivaKumarKalvaTR** [comment]
+
